core/src/main/scala/com/avast/grpc/jsonbridge/GrpcJsonBridgeBase.scala

```diff
@@ -1,27 +1,32 @@
 package com.avast.grpc.jsonbridge
 
+import cats.effect.{Async, Effect}
+import cats.syntax.all._
 import com.avast.grpc.jsonbridge.GrpcJsonBridge.GrpcHeader
 import com.google.protobuf.Message
 import com.google.protobuf.util.JsonFormat
 import com.typesafe.scalalogging.StrictLogging
+import io.grpc._
 import io.grpc.stub.MetadataUtils
-import io.grpc.{Metadata, Status, StatusException, StatusRuntimeException}
 import monix.eval.Task
+import monix.execution.Scheduler
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.language.higherKinds
 import scala.util.control.NonFatal
 
 /** This is trait for internal usage. You should not use it directly.
   */
-trait GrpcJsonBridgeBase[Stub <: io.grpc.stub.AbstractStub[Stub]] extends StrictLogging {
+trait GrpcJsonBridgeBase[F[_], Stub <: io.grpc.stub.AbstractStub[Stub]] extends StrictLogging {
 
   protected def newFutureStub: Stub
+  protected implicit def F: Async[F]
```

Comment:
What is pls purpose of this? I would expect this constraint defined in the trait header, like `GrpcJsonBridgeBase[F[_] <: Async]`

Reply:
It could be a constraint if it's not a trait :-) I can rework it if you insist but it would still be very similar and didn't bring anything.

Reply:
So is it a Scala limitation that _my_ syntax cannot be used on traits? Then I vote for changing this trait to class. There is only one implementation (in the macro) so it's not an issue IMHO.